src/main/resources/org/terasology/launcher/packages/schema.json

```diff
@@ -0,0 +1,34 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "type": "array",
+  "items": {
+    "type": "object",
+    "properties" : {
+
+      "url": {"type": "string", "format": "uri"},
+
+      "type": {"type": "string", "minLength": 1},
+
+      "trackedPackages": {
+        "type": "array",
+        "items": {
+          "oneOf": [
+
+            {"type": "string", "minLength": 1},
+
+            {
+              "type": "object",
+              "properties": {
+                "id": {"type": "string", "minLength": 1},
+                "name": {"type": "string", "minLength": 1}
+              },
+              "required": ["id", "name"]
+            }
```

Comment:
Let's have an individual schema.json for each option here.
At the latest, having only one will become very chaotic once we merge everything into a single `config.json`
This way we can also always check the most recent first and decide how to proceed afterwards. Ideally (stretch goal), we would even have a transformator from an older schema to the most recent one.

Reply:
I'm aware that this would result in sources.json files with mixed-schema entries not being readable anymore but tbh we wouldn't want those to lie around anyways, would we?